sycl/include/sycl/reduction.hpp

```diff
@@ -917,81 +918,81 @@ bool reduCGFuncForRangeFastReduce(handler &CGH, KernelType KernelFunc,
   accessor PartialSums(PartialSumsBuf, CGH, sycl::read_write, sycl::no_init);
 
   bool IsUpdateOfUserVar = !Reduction::is_usm && !Redu.initializeToIdentity();
-  using InitName =
-      __sycl_reduction_kernel<reduction::init_krn::GroupCounter, KernelName>;
-
-  // Integrated/discrete GPUs have different faster path.
-  auto NWorkGroupsFinished =
-      getDeviceFromHandler(CGH).get_info<info::device::host_unified_memory>()
-          ? Redu.getReadWriteAccessorToInitializedGroupsCounter(CGH)
-          : Redu.template getGroupsCounterAccDiscrete<InitName>(CGH);
-
-  auto DoReducePartialSumsInLastWG =
-      Reduction::template getReadWriteLocalAcc<int>(1, CGH);
-
-  using Name =
-      __sycl_reduction_kernel<reduction::main_krn::RangeFastReduce, KernelName>;
-  size_t PerGroup = Range.size() / NWorkGroups;
-  CGH.parallel_for<Name>(NDRange, [=](nd_item<1> NDId) {
-    // Call user's functions. Reducer.MValue gets initialized there.
-    typename Reduction::reducer_type Reducer;
-    reductionLoop(Range, PerGroup, Reducer, NDId, KernelFunc);
+  auto Rest = [&](auto NWorkGroupsFinished) {
```

Comment:
A little bit of a nit, but could this be moved to right before it is needed? Makes it a little easier to read the part that uses it and it may also avoid the creation of the object if we do an early exit before it is used.

Reply:
My bad. I thought this lambda ended a lot ealier than it did.